crates/vision/src/ball_detection.rs

```diff
@@ -332,18 +358,40 @@ fn project_balls_to_ground(
     clusters: &[BallCluster],
     camera_matrix: &CameraMatrix,
     ball_radius: f32,
-) -> Vec<Ball> {
+    measurement_noise: Vector2<Pixel>,
+    filtered_gyro_velocity: f32,
+) -> Vec<BallMeasurement> {
     clusters
         .iter()
         .filter_map(|cluster| {
-            let position = point![cluster.circle.center.x(), cluster.circle.center.y()];
-            match camera_matrix.pixel_to_ground_with_z(position, ball_radius) {
-                Ok(position) => Some(Ball {
-                    position,
-                    image_location: cluster.circle,
-                }),
-                Err(_) => None,
+            let position = camera_matrix
+                .pixel_to_ground_with_z(
+                    point![cluster.circle.center.x(), cluster.circle.center.y()],
+                    ball_radius,
+                )
+                .ok()?;
+            let projected_noise = {
+                let noise_in_image: Vector2<Pixel> = vector![
+                    measurement_noise.x() * (1.0 + filtered_gyro_velocity.abs()),
+                    measurement_noise.y() * (1.0 + filtered_gyro_velocity.abs())
```

Comment:
I think it makes sense to weigh x and y differently. If the robot only wobbles back and forth this would also increase the lateral noise in the same way, which doesn't make sense I think 🤔

Reply:
I removed this from this PR since I think it deserves its own